backend/src/helpers/walkRecursive.js

```diff
@@ -2,24 +2,26 @@
  * iterate through all fields and replace it with the callback result
  * @property data Array
  * @property fields Array
+ * @property fieldName String
  * @property callback Function
  */
-function walkRecursive(data, fields, callback, _key) {
+function walkRecursive(data, fields, fieldName, callback, _key) {
   if (!Array.isArray(fields)) {
     throw new Error('please provide an fields array for the walkRecursive helper')
   }
   if (data && typeof data === 'string' && fields.includes(_key)) {
     // well we found what we searched for, lets replace the value with our callback result
-    data = callback(data, _key)
+    const key = _key.split('!')
+    if (key.length === 1 || key[1] !== fieldName) data = callback(data, key[0])
```

Comment:
I do not understand `key[1] !== fieldName` shouldnt't it be `key[0] !== fieldName` or have `callback(data, key.length === 1 ? key[0] : key[1])

Reply:
It is `description!embed`.

Reply:
And I want to pass `description` to the callback